cmd/skaffold/app/cmd/config/util.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const defaultConfigLocation = ".skaffold/config"
+
+func resolveKubectlContext() {
+	if kubecontext != "" {
+		return
+	}
+
+	context, err := context.CurrentContext()
+	if err != nil {
+		logrus.Warn(errors.Wrap(err, "retrieving current kubectl context"))
+		kubecontext = "default"
+	}
+	if context == "" {
+		logrus.Infof("no context currently set, falling back to default")
+		kubecontext = "default"
+	}
+	kubecontext = context
+}
+
+func resolveConfigFile() error {
+	if configFile != "" {
+		// we had a config provided as a flag, expand it and return
+		if !filepath.IsAbs(configFile) {
```

Comment:
why does this need to be an absolute path? add the reasoning as a comment?

Reply:
thanks, this was from some previous logic which I removed a while back so I'll take this out